app/src/main/res/values-zh-rCN/strings.xml

```diff
@@ -1,457 +1,539 @@
-<?xml version='1.0' encoding='UTF-8'?>
+<?xml version="1.0" encoding="utf-8"?>
 <resources>
```

Comment:
Please update your branch and test if it works to manually select the simplified chinese through settings -> content -> app language.
If not, edit this line https://github.com/TeamNewPipe/NewPipe/blob/dev/app/src/main/res/values/settings_keys.xml#L992, change `zh-hans` into `zh-cn` and test again, it should be fine

Reply:
O K

Reply:
It works because ``b+zh+HANS+CN`` is still there. But it will select the file in ``b+zh+HANS+CN``, and not in ``zh-CN``.

I don't understand why do you keep ``b+zh+HANS+CN``, because if I understood the reason of the PR, ``b+zh+HANS+CN`` is a duplicate of ``zh-CN``? If ``b+zh+HANS+CN`` is the same as ``zh-CN``, you should delete the folder, because Weblate will still update ``b+zh+HANS+CN``, not ``zh-CN``, and also change [zh-hans here into zh-cn](https://github.com/TeamNewPipe/NewPipe/blob/dev/app/src/main/res/values/settings_keys.xml#L992)

Reply:
That is a historic problem.

Reply:
> It works because `b+zh+HANS+CN` is in the value folder. But it will select the file in `b+zh+HANS+CN`, and not in `zh-CN`.
> 
> I don't understand why do you keep `b+zh+HANS+CN`, because if I understood the reason of the PR, `b+zh+HANS+CN` is a duplicate of `zh-CN`? If `b+zh+HANS+CN` is the same as `zh-CN`, you should delete the folder, because Weblate will still update `b+zh+HANS+CN`, not `zh-CN`, and also change [zh-hans here into zh-cn](https://github.com/TeamNewPipe/NewPipe/blob/dev/app/src/main/res/values/settings_keys.xml#L992)

I'm afraid that it will influence Weblate.

Reply:
> I'm afraid that it will influence Weblate.

But that's the goal, all translations have to be done through Weblate. You won't be able to make PRs every updates to update zh-CN. So it has to be fixed once and then use Weblate.

Reply:
> > I'm afraid that it will influence Weblate.
> 
> But that's the goal, all translations have to be done through Weblate. You won't be able to make PRs every updates to update zh-CN. So it has to be fixed once and then use Weblate.

The problems I(we) faced is:
there are two various that take effect—zh-rCN(this file),b+zh-hans-cn . Weblate always updated the first one, b+zh-hans-cn won't be updated,and is old.
so, I moved b+zh-hans-cn to zh-rCN. if merging successfully, then delete b+zh-hans-cn (also try to let Weblate update zh-rCn ,so I(we) can directly translate Newpipe into Chinese)

BTW, does every language in Weblate have a source file. and Weblate just edit them?

Reply:
> 
> btw, can you check if [these lines](https://github.com/TeamNewPipe/NewPipe/blob/dev/app/src/main/res/values/settings_keys.xml#L1056-L1058) are good? The first should be traditionnal chinese, the second simplified chinese, the third hong-kong chinese, then cantonese I guess.

Yes , but we usually don't say them in our language.
And the " Hong Kong Chinese" is also called as Cantonese!

Reply:
> > btw, can you check if [these lines](https://github.com/TeamNewPipe/NewPipe/blob/dev/app/src/main/res/values/settings_keys.xml#L1056-L1058) are good? The first should be traditionnal chinese, the second simplified chinese, the third hong-kong chinese, then cantonese I guess.
> 
> Yes , but we usually don't say them in our language.
> And the " Hong Kong Chinese" is also called as Cantonese!

No, the Lang code was wrong.